### PR TITLE
Upgrade to Material UI v5

### PR DESCRIPTION
**Pull request by stbenjam**
We're currently on v4, which will go EOL when v6 comes out (soon).  This updates Sippy to material UI v5, working through https://mui.com/material-ui/migration/migration-v4/ and addressing other bugs as I find them.

Collecting issues here:
- [X] Update deps to mui v5
- [X] Run codemod fixes
- [X] Theme colors have changed 
- [X] Sidebar doesn't work anymore
- [x] Fix component readiness side bar - it's totally wrong on the first page load, and then looks fine on refresh
- [x] Job analysis page doesn't load at all
- [x] Component readiness tables are off to the side
- [x] Release overview page doesn't get centered
- [x] Test table is off the borders of the page
- [x] Tests-by-variage page (i.e. install) has vertical column headers?? http://localhost:3000/sippy-ng/install/4.15/operators
- [x] Margins on in-page tabs are way off e.g. [this page](http://localhost:3000/sippy-ng/jobs/4.15?filters=%7B%22items%22%3A%5B%7B%22columnField%22%3A%22current_runs%22%2C%22operatorValue%22%3A%22%3E%3D%22%2C%22value%22%3A%227%22%7D%2C%7B%22columnField%22%3A%22variants%22%2C%22operatorValue%22%3A%22contains%22%2C%22value%22%3A%22never-stable%22%2C%22not%22%3Atrue%7D%5D%7D&sortField=net_improvement&sort=asc)
- [x] Jobs page icons have blue hover (underlined from codemods?)
- [x] date pickers are broken (and component readiness won't load)

### CODE REVIEW

**Comment by neisw** [comment]
Picture being worth 1k words, I see multiple 'hamburger' menues.  Generat Report is missing background color and the group by / exclude dropdowns the background doesn't extend for the entire control, just the length of the text.  See similar sizing / placement issues with Jobs, Tests controls at the top like all tests, tests by variant, etc.

image

**Comment by stbenjam** [comment]
Thanks.  When you refresh component readiness does it fix itself? It did for me.  It's very odd :-(

**Comment by neisw** [comment]
Yep it did.  But not the Jobs or Tests controls at the top of those pages

image

**Comment by DennisPeriquet** [comment]
Jobs, try to make a filter and see the formatting of the list does not have new lines after each choice:

Screenshot 2023-11-14 at 12 07 31 PM

**Comment by stbenjam** [comment]
@DennisPeriquet Can you try the filters with the latest?

**Comment by DennisPeriquet** [comment]
similar problem:

Screenshot 2023-11-14 at 1 24 57 PM

**Comment by stbenjam** [comment]
@DennisPeriquet Hm ok, thanks.  It's also doing it in the other filter menus

**Comment by stbenjam** [comment]
@DennisPeriquet @neisw I think those are fixed, can you look again?

I think this is ready if you don't find anything major

**Comment by DennisPeriquet** [comment]
navigate to here: http://localhost:3000/sippy-ng/release/4.15/tags/4.15.0-0.ci-2023-11-02-230151 and then click the "PULL REQUESTS" on the top -- you get `bundle.js:25262 Uncaught ReferenceError: useTheme is not defined`

**Comment by stbenjam** [comment]
It works for me, did you run `npm install` again? You might also need to restart `npm start`

File: sippy-ng/src/releases/ReleasePayloadPullRequests.js [hunk 1]
```diff
@@ -1,30 +1,27 @@
-import { createTheme, makeStyles } from '@material-ui/core/styles'
-import { DataGrid } from '@material-ui/data-grid'
+import { DataGrid } from '@mui/x-data-grid'
+import { makeStyles } from '@mui/styles'
```
Comment (DennisPeriquet) [inline sippy-ng/src/releases/ReleasePayloadPullRequests.js:2]: if I make this `import { makeStyles } from '@mui/styles'`, clicking on "PULL REQUESTS" works; I see `useTheme()` used on line 23 in this file.

**Comment by stbenjam** [comment]
You are right, sorry, I missed the part about it being on the pull requests page. Fixed!

File: sippy-ng/src/releases/ReleasePayloadPullRequests.js (hunk 1 above)
Comment (stbenjam) [reply sippy-ng/src/releases/ReleasePayloadPullRequests.js]: You are right, sorry, I missed the part about it being on the pull requests page. Fixed!

**Comment by DennisPeriquet** [comment]
cancel hold when you're ready

**Comment by DennisPeriquet** [comment]
install page looks good too

**Pull request merged**